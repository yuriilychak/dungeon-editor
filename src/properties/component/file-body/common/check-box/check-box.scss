@import "../../../../../sizes";

.check-box-wrapper {
    display: flex;
    margin-right: $indent_l;
}

.check-box-icon-root {
    padding: 0 $indent_s 0 0 !important;
}

.check-box-label-root {
    margin-left: 0 !important;
    margin-right: 0 !important;
}

.check-box-label-label {
    font-size: $font-size_small !important;
}
