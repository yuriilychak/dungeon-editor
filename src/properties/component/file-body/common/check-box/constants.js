const stylePrefix = "check-box-";

export const STYLES = {
    ROOT: `${stylePrefix}wrapper`,
    LABEL: {
        label: `${stylePrefix}label-label`,
        root: `${stylePrefix}label-root`
    },
    CHECK_BOX: {
        root: `${stylePrefix}icon-root`
    }

};
