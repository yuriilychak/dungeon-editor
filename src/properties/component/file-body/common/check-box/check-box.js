import React, { memo } from "react";
import { string, bool, func } from "prop-types";

import Checkbox from "@material-ui/core/Checkbox";
import FormControlLabel from "@material-ui/core/FormControlLabel";

import { STYLES } from "./constants";
import { useChange } from "../../../../../hooks";

import "./check-box.scss";

export const CheckBox = ({ value, onChange, label }) => {
    const handleChange = useChange(onChange, event => event.target.checked);
    return (
        <div className={STYLES.ROOT}>
            <FormControlLabel
                classes={STYLES.LABEL}
                control={
                    <Checkbox
                        classes={STYLES.CHECK_BOX}
                        checked={value}
                        onChange={handleChange}
                        color="primary"
                    />}
                label={label}
            />
        </div>
    );
};

CheckBox.propTypes = {
    value: bool.isRequired,
    label: string.isRequired,
    onChange: func.isRequired
};

export default memo(CheckBox);
