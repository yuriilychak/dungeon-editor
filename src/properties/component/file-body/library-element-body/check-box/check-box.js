import React, { memo } from "react";
import Checkbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';

import "./check-box.css";

const CheckBox = memo(({ checked, onChange, label }) => (
    <div className="check-box-wrapper">
        <FormControlLabel
            control={
                <Checkbox
                    checked={checked}
                    onChange={onChange}
                    color="primary"
                />}
            label={label}
        />
    </div>
));

export default CheckBox;
