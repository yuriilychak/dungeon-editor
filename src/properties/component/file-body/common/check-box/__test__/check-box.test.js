import { createMount } from "@material-ui/core/test-utils";
import React from "react";

import { CheckBox } from "../check-box";

jest.mock("@material-ui/core/Checkbox", () => ({ onChange }) => (
    <button
        onClick={() => onChange({ target: { checked: true } })}
    >
        Mock Csheckbox
    </button>
));

describe("GIVEN Checkbox from property section", () => {
    const mockChange = jest.fn();
    const wrapper = createMount()(
        <CheckBox
            value={true}
            label="test"
            onChange={mockChange}
        />
    );

    it("WHEN render component with default props THEN component should match snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("WHEN dispatch onChange THAN result should equal expected object", () => {
        wrapper.find("button").simulate("click");
        expect(mockChange).toHaveBeenCalledWith(true);
    });
});
