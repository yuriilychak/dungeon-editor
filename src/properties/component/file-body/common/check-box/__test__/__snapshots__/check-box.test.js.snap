// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN Checkbox from property section WHEN render component with default props THEN component should match snapshot 1`] = `"<div class=\\"check-box-wrapper\\"><label class=\\"MuiFormControlLabel-root check-box-label-root\\"><button>Mock Csheckbox</button><span class=\\"MuiTypography-root MuiFormControlLabel-label check-box-label-label MuiTypography-body1\\">test</span></label></div>"`;
